docs(other): respect usage article (#149)
Redocly/website: learn/arazzo/practical-example-series/api-contract-testing-01.md

{% code-walkthrough
  filesets=[
    {
      "files": [
        "../_filesets/practical-example-series/contract-testing/redocly-cafe-api.arazzo.yaml",
        "../_filesets/practical-example-series/contract-testing/redocly-cafe-api.yaml"
      ]
    }
  ]
%}

> # Respect Practical Example Series: API Contract Testing with Respect

Reviewer: ```suggestion
# Test API contracts using Redocly Respect
```

> This article shows how to use Respect, powered by Arazzo workflows, for API contract testing.

Reviewer: ```suggestion
Use Respect, powered by Arazzo workflows, for API contract testing.
```

> You will cover the following topics:

Reviewer: ```suggestion
Learn how to:
```

- Automating repetitive API workflows.
- Adding API contract tests to CI/CD routines.
- Keeping API documentation synchronized with actual API behavior.
- Sharing described workflows across teams.

> - Using the open-source [Redocly CLI](https://www.npmjs.com/package/@redocly/cli) to execute Arazzo workflows, inspect contract test results, and fix mismatches between the OpenAPI description and the actual API response.

Reviewer: ```suggestion
- Automate repetitive API workflows.
- Add API contract tests to CI/CD routines.
- Keep API documentation synchronized with actual API behavior.
- Share described workflows across teams.
- Use the open-source [Redocly CLI](https://www.npmjs.com/package/@redocly/cli) to execute Arazzo workflows, inspect contract test results, and fix mismatches between the OpenAPI description and the actual API response.
```

## The problem

Many teams test APIs by writing test code in the same language as the API implementation.
This can work well for a single service, but it becomes harder to maintain in systems with multiple products, repositories, and programming languages.

Another common challenge is keeping API documentation aligned with actual API behavior.
Developers may update an endpoint but forget to update its OpenAPI description, or the documentation may change without the implementation following it.
Over time, these small differences make it harder for QA engineers, developers, and technical writers to trust the API contract.

Respect helps address this problem by executing Arazzo workflows against a running API and validating the responses against the connected OpenAPI description.

> This makes API contract testing more declarative and easier to share across teams.

Reviewer: ```suggestion
This strategy makes API contract testing more declarative and easier to share across teams.
```

## Prerequisites

To follow the examples in this article, you need:

> - Basic familiarity with [Arazzo](https://redocly.com/learn/arazzo/what-is-arazzo).

Reviewer: ```suggestion
- Basic familiarity with [Arazzo](../what-is-arazzo).
```

> - An API described with OpenAPI. The examples use a modified version of the Redocly Cafe API description.

Reviewer: ```suggestion
- An API described with OpenAPI.
    The examples use a modified version of the Redocly Cafe API description.
```

> > **Important:** The API description used here intentionally contains a discrepancy for demonstration purposes.

Reviewer: ```suggestion
{% admonition type="info" name="Tutorial content" %}
 The API description used here intentionally contains a discrepancy for demonstration purposes.
{% /admonition %}
```

{% step id="openapi-list-menu-items" heading="GET /menu operation" %}
The `GET /menu` operation returns a paginated list of menu items.
The `200` response points to the `MenuItemList` schema, which is where the intentional discrepancy lives.
{% /step %}

{% step id="openapi-broken-schema" heading="MenuItemList (broken)" %}
The schema declares the response as an **array**, but the real API returns a **paginated object** with `object`, `page`, and `items` fields.
Later in this walkthrough, Respect will compare this contract with the real API response and surface the mismatch.
{% /step %}

## Create an Arazzo description

There are several ways to create an Arazzo description:

- Read the [Arazzo specification](https://spec.openapis.org/arazzo/latest.html) and write the file from scratch for full control.
- Use the `npx @redocly/cli generate-arazzo openapi.yaml` command as a starting point.
- Use AI assistance to draft the Arazzo description, then lint the output to catch structural or syntax errors.

Whichever approach you choose, validate the Arazzo file with Redocly CLI:

```bash
npx @redocly/cli lint redocly-cafe-api.arazzo.yaml
```

> The tool will let you know that the description is valid.

Reviewer: ```suggestion
The tool informs you that the description is valid.
```

```yaml
MenuItemList:
  type: object
  properties:
    object:
      type: string
      const: list
      description: Entity name.
    page:
      $ref: '#/components/schemas/Page'
    items:
      type: array
      items:
        $ref: '#/components/schemas/MenuItem'
  required:
    - object
    - page
    - items
```
{% /step %}

Re-run the workflow to confirm that all checks now pass:

```bash
npx @redocly/cli respect redocly-cafe-api.arazzo.yaml
```

After the schema is updated, the response body matches the documented contract.
This demonstrates the core value of contract testing with Respect: when the API and its OpenAPI description drift apart, the workflow highlights the mismatch immediately.

## Practical applications

After you create and verify an Arazzo workflow locally, you can use it in several ways:

- Include the workflow in a CI/CD pipeline to keep API documentation synchronized with actual API behavior:

  ```bash
  # Spawn your API instance, then run the workflow
  npx @redocly/cli respect redocly-cafe-api.arazzo.yaml --verbose
  ```

- Automate routine API workflows for development and QA tasks.

> - Describe important application flows with Arazzo and share them with team members. Non-technical users can also use visualization tools like Replay to understand the workflows.

Reviewer: ```suggestion
- Describe important application flows with Arazzo and share them with team members.
    Non-technical users can also use visualization tools like Replay to understand the workflows.
```

{% code-walkthrough
  filesets=[
    {
      "files": [
        "../_filesets/practical-example-series/contract-testing/redocly-cafe-api.arazzo.yaml",
        "../_filesets/practical-example-series/contract-testing/redocly-cafe-api.yaml"
      ]
    }
  ]
%}

# Test API contracts using Redocly Respect

Use Respect, powered by Arazzo workflows, for API contract testing.

Learn how to:

- Automate repetitive API workflows.
- Add API contract tests to CI/CD routines.
- Keep API documentation synchronized with actual API behavior.
- Share described workflows across teams.
- Use the open-source [Redocly CLI](https://www.npmjs.com/package/@redocly/cli) to execute Arazzo workflows, inspect contract test results, and fix mismatches between the OpenAPI description and the actual API response.

## The problem

Many teams test APIs by writing test code in the same language as the API implementation.

> This can work well for a single service, but it becomes harder to maintain in systems with multiple products, repositories, and programming languages.

Reviewer: ```suggestion
This approach can work well for a single service, but it becomes harder to maintain in systems with multiple products, repositories, and programming languages.
```

The `GET /menu` operation returns a paginated list of menu items.
The `200` response points to the `MenuItemList` schema, which is where the intentional discrepancy lives.
{% /step %}

{% step id="openapi-broken-schema" heading="MenuItemList (broken)" %}
The schema declares the response as an **array**, but the real API returns a **paginated object** with `object`, `page`, and `items` fields.
Later in this walkthrough, Respect will compare this contract with the real API response and surface the mismatch.
{% /step %}

## Create an Arazzo description

There are several ways to create an Arazzo description:

- Read the [Arazzo specification](https://spec.openapis.org/arazzo/latest.html) and write the file from scratch for full control.
- Use the `npx @redocly/cli generate-arazzo openapi.yaml` command as a starting point.
- Use AI assistance to draft the Arazzo description, then lint the output to catch structural or syntax errors.

Whichever approach you choose, validate the Arazzo file with Redocly CLI:

```bash
npx @redocly/cli lint redocly-cafe-api.arazzo.yaml
```

The tool informs you that the description is valid.

> Now switch to `redocly-cafe-api.arazzo.yaml` in the right panel and walk through it section by section.

Reviewer: ```suggestion
Switch to `redocly-cafe-api.arazzo.yaml` in the right panel and walk through it section by section.
```

This is the intentional discrepancy in the example.

Respect made this drift visible immediately, without anyone having to read both the OpenAPI description and the API output side-by-side.
{% /step %}

## Fix the schema

{% step id="schema-fix" heading="Updated MenuItemList schema" %}
In `redocly-cafe-api.yaml`, change `MenuItemList` from an `array` to an `object` with `object`, `page`, and `items` properties so the schema matches the response the API actually returns.

Compare the highlighted block in the right panel with the corrected version below:

```yaml
MenuItemList:
  type: object
  properties:
    object:
      type: string
      const: list
      description: Entity name.
    page:
      $ref: '#/components/schemas/Page'
    items:
      type: array
      items:
        $ref: '#/components/schemas/MenuItem'
  required:
    - object
    - page
    - items
```
{% /step %}

Re-run the workflow to confirm that all checks now pass:

```bash
npx @redocly/cli respect redocly-cafe-api.arazzo.yaml
```

After the schema is updated, the response body matches the documented contract.

> This demonstrates the core value of contract testing with Respect: when the API and its OpenAPI description drift apart, the workflow highlights the mismatch immediately.

Reviewer: ```suggestion

This example demonstrates the core value of contract testing with Respect: when the API and its OpenAPI description drift apart, the workflow highlights the mismatch immediately.
```

{% code-walkthrough
  filesets=[
    {
      "files": [
        "../_filesets/practical-example-series/contract-testing/redocly-cafe-api.arazzo.yaml",
        "../_filesets/practical-example-series/contract-testing/redocly-cafe-api.yaml"
      ]
    }
  ]
%}

# Respect Practical Example Series: API Contract Testing with Respect

This article shows how to use Respect, powered by Arazzo workflows, for API contract testing.
You will learn how to describe an API workflow, execute it with Redocly CLI, and use the results to find mismatches between an OpenAPI description and the real API behavior.

You will cover the following topics:

Practical applications of Arazzo:

- Automating repetitive API workflows.
- Adding API contract tests to CI/CD routines.
- Keeping API documentation synchronized with actual API behavior.
- Sharing described workflows across teams.

> - Using the open-source [@redocly/cli](https://www.npmjs.com/package/@redocly/cli) `respect` command to execute Arazzo workflows, inspect contract test results, and fix mismatches between the OpenAPI description and the actual API response.

Reviewer 2: ```suggestion
This article shows how to use Respect, powered by Arazzo workflows, for API contract testing.
 
You will cover the following topics:
 
- Automating repetitive API workflows.
- Adding API contract tests to CI/CD routines.
- Keeping API documentation synchronized with actual API behavior.
- Sharing described workflows across teams.
- Using the open-source Redocly CLI to execute Arazzo workflows, inspect contract test results, and fix mismatches between the OpenAPI description and the actual API response.
```

BTW, This is a lot of topics. Maybe split them into two articles?

Author: All of those needed to cover flow with Auth, but we can make smaller about reusable Arazzo.

## The problem

Many teams test APIs by writing test code in the same language as the API implementation.
This can work well for a single service, but it becomes harder to maintain in systems with multiple products, repositories, and programming languages.

Another common challenge is keeping API documentation aligned with actual API behavior.
Developers may update an endpoint but forget to update its OpenAPI description, or the documentation may change without the implementation following it.
Over time, these small differences make it harder for QA engineers, developers, and technical writers to trust the API contract.

Respect helps address this problem by executing Arazzo workflows against a running API and validating the responses against the connected OpenAPI description.
This makes API contract testing more declarative and easier to share across teams.

## Prerequisites

To follow the examples in this article, you need:

> - Basic familiarity with [Arazzo](../what-is-arazzo.md).

Reviewer 2: Let's add a link to this article from '../what-is-arazzo.md'?

Author: But it is a link to the md, or would you prefer URL?

{% step id="openapi-list-menu-items" heading="GET /menu operation" %}
The `GET /menu` operation returns a paginated list of menu items.
The `200` response points to the `MenuItemList` schema, which is where the intentional discrepancy lives.
{% /step %}

{% step id="openapi-broken-schema" heading="MenuItemList (broken)" %}
The schema declares the response as an **array**, but the real API returns a **paginated object** with `object`, `page`, and `items` fields.
Later in this walkthrough, Respect will compare this contract with the real API response and surface the mismatch.
{% /step %}

## Create an Arazzo description

There are several ways to create an Arazzo description:

- Read the [Arazzo specification](https://spec.openapis.org/arazzo/latest.html) and write the file from scratch for full control.
- Use the `npx @redocly/cli generate-arazzo openapi.yaml` command as a starting point.
- Use AI assistance to draft the Arazzo description, then lint the output to catch structural or syntax errors.

Whichever approach you choose, validate the Arazzo file with Redocly CLI:

```bash

> npx @redocly/cli@latest lint redocly-cafe-api.arazzo.yaml

Reviewer 2: ```suggestion
npx @redocly/cli lint redocly-cafe-api.arazzo.yaml
```

Let's save some space...

```

> ```bash

Reviewer 2: I'd not put several bash sections one after another. Either add a screenshot or just say something like this:

```
The tool will let you know that the description is valid.
```

No configurations were provided -- using built in recommended configuration by default.

validating redocly-cafe-api.arazzo.yaml...
redocly-cafe-api.arazzo.yaml: validated in 6ms

Woohoo! Your API description is valid. 🎉
```

Now switch to `redocly-cafe-api.arazzo.yaml` in the right panel and walk through it section by section.

{% step id="arazzo-version" heading="Declare the Arazzo version" %}
Every Arazzo file begins with a version declaration.

> The latest stable version is `1.0.1`.

Reviewer 2: This is definitely redundant if you have basic familiarity with Arazzo. Either remove this or do add a separate guide on creating the Arazzo description itself.

Author: I removed some, but maybe we can keep at least pointers started from workflow level?

      keep-alive: timeout=5
      vary: Origin
      x-powered-by: Express
    Response Size: 435 bytes
    Response Body:
      {
        "object": "list",
        "page": {
          "startCursor": "ixCALWlkOnByZF8wMDAwMDAwMDAwc2VlZHRyYW1zMDAwMDAwMAM",
          "endCursor": "ixCALWlkOnByZF8wMDAwMDAwMDAwc2VlZHRyYW1zMDAwMDAwMAM",
          "hasNextPage": true,
          "hasPrevPage": false,
          "limit": 1,
          "total": 5
        },
        "items": [
          {
            "id": "prd_0000000000seedtrams0000000",
            "name": "tiramisu",
            "price": 13000,
            "category": "dessert",
            "createdAt": "2026-04-29T10:00:50.610Z",
            "updatedAt": "2026-04-29T10:00:50.610Z",
            "object": "menuItem",
            "calories": 450
          }
        ]
      } 
 
    ✓ success criteria check - $statusCode == 200
    ✓ status code check - $statusCode in [200, 400, 500]
    ✓ content-type check
    ✗ schema check 

  Summary for redocly-cafe-api.arazzo.yaml
  
  Workflows: 1 failed, 1 total
  Steps: 1 failed, 1 total
  Checks: 3 passed, 1 failed, 4 total
  Time: 58ms 
```

</details>

> In addition to the success criterion declared in the Arazzo file, Respect performed three additional checks automatically:

Reviewer 2: You've already said this here: https://github.com/Redocly/website/pull/149/changes#diff-90b9281218ab219b15a4758675d34ce45fe04dfde843f556a82dd8ade246a351R140. Please avoid repeating yourself.

Author: Its more detailed description compare to 
```
Respect also runs additional automatic checks based on the connected OpenAPI description (see the next section).
```

Reviewer: I'd leave the text as it is.
The tutorial is targeted to beginner users, so it shouldn't be a problem.

- `status code check`
- `content-type check`
- `schema check`

Because the workflow step is connected to an OpenAPI description, [@redocly/cli](https://www.npmjs.com/package/@redocly/cli) verifies that the API returns a documented status code, the expected content type, and a response body that matches the documented schema.

## Understand the schema failure

Switch back to `redocly-cafe-api.yaml` in the right panel.
The failed schema check reports the following message:

```text
| ^^ 👈🏽  type must be array
```

{% step id="schema-failure" heading="Spot the mismatch" %}
The OpenAPI description declares `MenuItemList` as an **array**, but the actual API response is a paginated **object** with `object`, `page`, and `items` properties.
This is the intentional discrepancy in the example.

Respect made this drift visible immediately, without anyone having to read both the OpenAPI description and the API output side-by-side.
{% /step %}

## Fix the schema

> In `redocly-cafe-api.yaml`, change `MenuItemList` from an `array` to an `object` with `object`, `page`, and `items` properties so the schema matches the response the API actually returns:

Reviewer 2: Let's link this as a step to `redocly-cafe-api.yaml`'s MenuItemList so we can see the difference between the schemas.

```yaml
MenuItemList:
  type: object
  properties:
    object:
      type: string
      const: list
      description: Entity name.
    page:
      $ref: '#/components/schemas/Page'
    items:
      type: array
      items:
        $ref: '#/components/schemas/MenuItem'
  required:
    - object
    - page
    - items
```

Re-run the workflow to confirm that all checks now pass:

```bash
npx @redocly/cli respect redocly-cafe-api.arazzo.yaml
```

After the schema is updated, the response body matches the documented contract.
This demonstrates the core value of contract testing with Respect: when the API and its OpenAPI description drift apart, the workflow highlights the mismatch immediately.

> ## Practical applications

Reviewer 2: I'm not even sure we have to explain practical cases in this article. I'd restrict that to saying that from now on it could be used in a CI/CD pipeline to check if there's no divergency between the docs and the server.

Author: cc: @Redocly/technical-writers

Reviewer: I'd leave the practical cases for now.

After you create and verify an Arazzo workflow locally, you can use it in several ways:

- Include the workflow in a CI/CD pipeline to keep API documentation synchronized with actual API behavior:

  ```bash
  # Spawn your API instance
  npm install @redocly/cli@latest -g

>   redocly respect products.arazzo.yaml --verbose

Reviewer 2: Let's not confuse readers with different running options. Let's use the `npx ...` way everywhere.

Reviewer 2: And that could be put into the Summary section anyway.

Author: Lets keep the Summary small.

Many teams test APIs by writing test code in the same language as the API implementation. This can work well for a single service, but it becomes harder to maintain in systems with multiple products, repositories, and programming languages.

Another common challenge is keeping API documentation aligned with actual API behavior. Developers may update an endpoint but forget to update its OpenAPI description, or the documentation may change without the implementation following it. Over time, these small differences make it harder for QA engineers, developers, and technical writers to trust the API contract.

Respect helps address this problem by executing Arazzo workflows against a running API and validating the responses against the connected OpenAPI description. This makes API contract testing more declarative and easier to share across teams.

### Prerequisites

To follow the examples in this article, you need:

- Familiarity with [Arazzo](../what-is-arazzo.md).
- An API described with OpenAPI. The examples use a modified version of the Redocly Cafe API.
  <b>IMPORTANT: the API description intentionally contains discrepancies for demonstration purposes.</b>

```yaml

> openapi: 3.1.0

Author: Using the 3.1.0 example, as VSCE currently doesn’t support rendering 3.2.0 and we would like to allow users to have valid preview.

# Respect Practical Example Series [01]

> ## API Contract Testing with Respect

Reviewer 2: ```suggestion
# Respect Practical Example Series: API Contract Testing with Respect
```

> This article shows how to use Respect, powered by Arazzo workflows, for API contract testing. You will learn how to describe an API workflow, execute it with Redocly CLI, and use the results to find mismatches between an OpenAPI description and the real API behavior.

Reviewer 2: ```suggestion
This article shows how to use Respect, powered by Arazzo workflows, for API contract testing.
You will learn how to describe an API workflow, execute it with Redocly CLI, and use the results to find mismatches between an OpenAPI description and the real API behavior.
```

Please use semantic breaks.

You will cover the following topics:

- Practical applications of Arazzo:
  - Automating repetitive API workflows.
  - Adding API contract tests to CI/CD routines.
  - Keeping API documentation synchronized with actual API behavior.
  - Sharing described workflows across teams.
- Using the open-source [@redocly/cli](https://www.npmjs.com/package/@redocly/cli) `respect` command to execute Arazzo workflows.

> ### What you will build

Reviewer 2: ```suggestion
## What you will build
```

No need in using the 3rd level here and further.

> You will create a simple Arazzo workflow that calls the `GET /menu` operation from an OpenAPI description. Then you will run the workflow with Respect, inspect the contract test results, and fix an intentional mismatch between the OpenAPI description and the actual API response.

Reviewer 2: Why saying 2 times what you'll do? Let's gather that in one place.

Author: lets move up

### The problem

Many teams test APIs by writing test code in the same language as the API implementation. This can work well for a single service, but it becomes harder to maintain in systems with multiple products, repositories, and programming languages.

Another common challenge is keeping API documentation aligned with actual API behavior. Developers may update an endpoint but forget to update its OpenAPI description, or the documentation may change without the implementation following it. Over time, these small differences make it harder for QA engineers, developers, and technical writers to trust the API contract.

Respect helps address this problem by executing Arazzo workflows against a running API and validating the responses against the connected OpenAPI description. This makes API contract testing more declarative and easier to share across teams.

### Prerequisites

To follow the examples in this article, you need:

> - Familiarity with [Arazzo](../what-is-arazzo.md).

Reviewer 2: ```suggestion
Respect helps address this problem by executing Arazzo workflows against a running API and validating the responses against the connected OpenAPI description. This makes API contract testing more declarative and easier to share across teams.

### Prerequisites

To follow the examples in this article, you need:

- Basic familiarity with [Arazzo](../what-is-arazzo.md).
```

The example uses the following OpenAPI description. The important part for this article is the `GET /menu` operation and its `MenuItemList` response schema.

<details>
<summary>OpenAPI description used in this example</summary>

```yaml
openapi: 3.1.0
info:
  title: Redocly Cafe
  description: |
    Demo API for cafe operators (not customers) to manage menus.
    Create API credentials and try it yourself in a realistic OpenAPI workflow.
  version: 1.0.0
  contact:
    email: team@redocly.com
    url: https://redocly.com/contact-us/
  license:
    name: MIT
    url: https://opensource.org/licenses/MIT
  termsOfService: https://redocly.com/subscription-agreement
servers:
  - url: https://cafe.cloud.redocly.com
    description: Live server.
tags:
  - name: Authorization
    description: Create a client to demo the API.
  - name: Products
    description: Operations related to products.
paths:
  /menu:
    get:
      tags:
        - Products
      summary: List all menu items
      description: Retrieve a collection of menu items with optional filtering and pagination.
      operationId: listMenuItems
      security: []

>       parameters:

Reviewer 2: Could we make the exemplary document  smaller, e.g. by removing parameters, redundant responses, examples etc.?

Reviewer 2: Let's focus on the Menu structure (as this is the main discrepancy).

Whichever approach you choose, validate the Arazzo file with Redocly CLI:

```bash
npx @redocly/cli@latest lint redocly-cafe-api.arazzo.yaml
```

```bash
No configurations were provided -- using built in recommended configuration by default.

validating redocly-cafe-api.arazzo.yaml...
redocly-cafe-api.arazzo.yaml: validated in 6ms

Woohoo! Your API description is valid. 🎉
```

The following simple workflow retrieves the menu items list.

`redocly-cafe-api.arazzo.yaml`

```yaml
arazzo: 1.0.1
info:
  title: Redocly Cafe API - Products
  version: 1.0.0
  description: This is the API for the Redocly Cafe Products.
sourceDescriptions:
  - name: redocly-cafe-api
    type: openapi
    url: redocly-cafe-api.yaml

workflows:
  - workflowId: menu-items-workflow
    summary: Menu Items Workflow
    steps:
      - stepId: get-products
        operationId: $sourceDescriptions.redocly-cafe-api.listMenuItems
        description: This step gets all products.
        parameters:
          - in: query
            name: limit
            value: 1
        successCriteria:
          - condition: $statusCode == 200
```

Key components:

> - `sourceDescriptions` define the connection to the OpenAPI description.

Reviewer 2: ```suggestion
- `sourceDescriptions` defines the connection to the OpenAPI description.
```

> - `sourceDescriptions` define the connection to the OpenAPI description.

Reviewer 2: ```suggestion
Please note that `sourceDescriptions` defines the connection to the OpenAPI description.
```

> ``` yaml

Reviewer 2: No need to repeat, you already declared this here: https://github.com/Redocly/website/pull/149/changes#diff-90b9281218ab219b15a4758675d34ce45fe04dfde843f556a82dd8ade246a351R411.

Author: Its for better readability, so that user won't scroll up.

sourceDescriptions:
  - name: redocly-cafe-api
    type: openapi
    url: redocly-cafe-api.yaml
```

> - Each step uses an `operationId` to connect the workflow step to an operation in the OpenAPI description.

Reviewer 2: ```suggestion
Each step uses an `operationId` to connect the workflow step to an operation in the OpenAPI description, like so:
```

Let's use more natural language rather than bullets. What do you think?

        - $ref: '#/components/parameters/Limit'
      responses:
        '200':
          description: Successful operation.
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/MenuItemList'
        '400':
          $ref: '#/components/responses/BadRequest'
        '500':
          $ref: '#/components/responses/InternalServerError'
```

This connection is important because Respect can use the OpenAPI description to verify the response status, content type, and schema.

- Steps can pass parameters to the operation. In this example, the `limit` query parameter restricts the response to one menu item.

```yaml
  parameters:
    - in: query
      name: limit
      value: 1
```

- Steps can also include a `successCriteria` section to define workflow-specific expectations.

```yaml
successCriteria:
  - condition: $statusCode == 200
```

### Execute the workflow using Redocly CLI

[@redocly/cli](https://www.npmjs.com/package/@redocly/cli) is an open-source tool that can execute Arazzo descriptions with the `respect` command.

```bash
npx @redocly/cli@latest respect redocly-cafe-api.arazzo.yaml
```

> 

Reviewer 2: Let's add something like this:

> Please notice that the workflow gets executed because we specified the live server in the OpenAPI description, so you don't have to specify that when writing tests.

The execution result looks like this. The full output is useful when debugging, but the most important lines are the checks near the end of the step:

- The success criteria check passes because the API returns status code `200`.
- The status code and content type checks pass because the response matches the documented response metadata.
- The schema check fails because the response body shape does not match the documented schema.

> <details>

Author: @Redocly/technical-writers is it better to use terminal output screenshot here?

Reviewer: I think it would be hard to fit the entire output on a screenshot below 100 kB.
I'd leave the text output.

        - object
        - page
        - items
```

Run the workflow again to confirm that all checks now pass.

```bash
npx @redocly/cli@latest respect redocly-cafe-api.arazzo.yaml
```

After the schema is updated, the response body matches the documented contract. This demonstrates the core value of contract testing with Respect: when the API and its OpenAPI description drift apart, the workflow highlights the mismatch immediately.

### Practical applications

After you create and verify an Arazzo workflow locally, you can use it in several ways:

- Include the workflow in a CI/CD pipeline to keep API documentation synchronized with actual API behavior:

```bash
# Spawn your API instance
npm install @redocly/cli@latest -g
redocly respect products.arazzo.yaml --verbose
```

- Automate routine API workflows for development and QA tasks.
- Describe important application flows with Arazzo and share them with team members. Non-technical users can also use visualization tools like Replay to understand the workflows.

### Summary

Arazzo provides a standard way to describe API workflows declaratively. Respect uses those workflows to run API contract tests against real API behavior.

> With this approach, teams can maintain API test coverage, detect documentation drift earlier, and share executable API workflows across engineering, QA, and documentation teams.

Reviewer 2: Do we want to put a link to the next part of this tutorial somewhere here?

Author: cc: @Redocly/technical-writers
